Mathlib/Algebra/GroupWithZero/WithZero.lean

```diff
@@ -230,16 +230,14 @@ instance involutiveInv [InvolutiveInv α] : InvolutiveInv (WithZero α) where
 instance divisionMonoid [DivisionMonoid α] : DivisionMonoid (WithZero α) where
   __ := divInvMonoid
   __ := involutiveInv
-  mul_inv_rev a b := match a, b with
+  mul_inv_rev
     | none, none => rfl
     | none, some _ => rfl
     | some _, none => rfl
     | some _, some _ => congr_arg some (mul_inv_rev _ _)
-  inv_eq_of_mul a b := match a, b with
-    | none, none => fun _ ↦ rfl
-    | none, some b => fun _ ↦ by contradiction
-    | some a, none => fun _ ↦ by contradiction
-    | some _, some _ => fun h ↦
+  inv_eq_of_mul
```

Comment:
Why does this work? Where do the other two cases "disappear"?

Reply:
In `fun _ ↦ by contradiction`, the `contradiction` was showing that the `_` was contradictory because it was a proof of `none = some _` (or `some _ = none`). But pattern matching already knows how to eliminate those contradictory branches, so there's no need to spell them out

Reply:
:bulb: Thanks!